CMakeLists.txt

```diff
@@ -1,41 +1,323 @@
   message(STATUS "Setting build type to 'Release' as none was specified.")
-  set(CMAKE_BUILD_TYPE Release CACHE STRING "Choose the type of build." FORCE)
+  set(CMAKE_BUILD_TYPE Release CACHE STRING 
+      "Choose the type of build, options are: Release|Debug|RelWithDebInfo (for distros)." FORCE)
 endif()
 
 message(STATUS "Install directory: ${CMAKE_INSTALL_PREFIX}")
+
 # About this project
 # ------------------------------------------------------------------------------
-project(moni)
-SET(VERSION_MAJOR "0")
-SET(VERSION_MINOR "1")
-SET(VERSION_PATCH "0")
-SET(VERSION "${VERSION_MAJOR}.${VERSION_MINOR}.${VERSION_PATCH}")
+project(moni-align)
+set(VERSION_MAJOR "0")
+set(VERSION_MINOR "1")
+set(VERSION_PATCH "0")
+set(VERSION "${VERSION_MAJOR}.${VERSION_MINOR}.${VERSION_PATCH}")
 
 # Set environment
 # ------------------------------------------------------------------------------
+find_package(Git REQUIRED)
+
+# Configure thirdparty
+# ------------------------------------------------------------------------------
+set(CMAKE_INSTALL_INCLUDEDIR "include") # This is an hack because include(GUIInstallDirs) doesn't work
 
-find_package(Git)
-if(GIT_FOUND)
-    message(STATUS "git found: ${GIT_EXECUTABLE}")
+# Set the output directories for executables and libraries
+if (NOT CMAKE_RUNTIME_OUTPUT_DIRECTORY)
+  message(STATUS "CMAKE_RUNTIME_OUTPUT_DIRECTORY not defined. Setting variable to ${CMAKE_CURRENT_BINARY_DIR}/bin")
+  set(CMAKE_RUNTIME_OUTPUT_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/bin)
+else()
+  message(STATUS "CMAKE_RUNTIME_OUTPUT_DIRECTORY user defined as ${CMAKE_RUNTIME_OUTPUT_DIRECTORY}")
+endif()
+if (NOT CMAKE_LIBRARY_OUTPUT_DIRECTORY)
+  message(STATUS "CMAKE_LIBRARY_OUTPUT_DIRECTORY not defined. Setting variable to ${CMAKE_CURRENT_BINARY_DIR}/lib")
+  set(CMAKE_LIBRARY_OUTPUT_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/lib)
+else()
+  message(STATUS "CMAKE_LIBRARY_OUTPUT_DIRECTORY user defined as ${CMAKE_LIBRARY_OUTPUT_DIRECTORY}")
+endif()
+if (NOT CMAKE_ARCHIVE_OUTPUT_DIRECTORY)
+  message(STATUS "CMAKE_ARCHIVE_OUTPUT_DIRECTORY not defined. Setting variable to ${CMAKE_LIBRARY_OUTPUT_DIRECTORY}")
+  set(CMAKE_ARCHIVE_OUTPUT_DIRECTORY ${CMAKE_LIBRARY_OUTPUT_DIRECTORY})
 else()
-    message(WARNING "git not found. Cloning of submodules will not work.")
+  message(STATUS "CMAKE_ARCHIVE_OUTPUT_DIRECTORY user defined as ${CMAKE_ARCHIVE_OUTPUT_DIRECTORY}")
 endif()
 
+include(AddGitSubmodule)
+include(FetchContent)
+include(ExternalProject)
 
+# Set up CMAKE variables that control where to look 
+list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_BINARY_DIR}") #find_package
+list(APPEND CMAKE_PREFIX_PATH "${CMAKE_LIBRARY_OUTPUT_DIRECTORY}" "${CMAKE_ARCHIVE_OUTPUT_DIRECTORY}") #find_library
 
-# Configure thirdparty
-# ------------------------------------------------------------------------------
-set(CMAKE_INSTALL_INCLUDEDIR "include") # This is an hack because include(GUIInstallDirs) doesn't work
+# ZLIB
+find_package(ZLIB REQUIRED)
+
+# SDSL
```

Comment:
Please move the thirdparty related components into a CMakeList.txt in the `thirdparty` folder so it is cleaner to read.